CreeDictionary/utils/cree_lev_dist.py

```diff
@@ -1,18 +1,35 @@
+import string
 import unicodedata
 
 VOWELS = {"a", "e", "i", "o"}
 
+_diacritic_letter_to_ascii = {
+    ascii_letter: ascii_letter for ascii_letter in string.ascii_lowercase
+}
+
+_diacritic_letter_to_ascii.update(
+    {"â": "a", "ā": "a", "ê": "e", "ē": "e", "ī": "i", "î": "i", "ô": "o", "ō": "o"}
+)
+
+_diacritic_letter_to_ascii.update(
+    {d.upper(): a.upper() for d, a in _diacritic_letter_to_ascii.items()}
+)
+
 
 def remove_cree_diacritics(input_str) -> str:
     """
     >>> remove_cree_diacritics('ā')
     'a'
     >>> remove_cree_diacritics('â')
     'a'
+    >>> remove_cree_diacritics('ÂĀÊĒ')
+    'AAEE'
+    >>> remove_cree_diacritics('âāêē')
+    'aaee'
+    >>> remove_cree_diacritics('âāêē âāêē')
+    'aaee aaee'
     """
-    nfkd_form = unicodedata.normalize("NFKD", input_str)
-    only_ascii = nfkd_form.encode("ASCII", "ignore").decode("utf-8")
-    return only_ascii
+    return "".join([_diacritic_letter_to_ascii.get(c, c) for c in input_str])
```

Comment:
replaced old magic with memorization based sweety

Reply:
You may want to look at [`str.maketrans()`](https://docs.python.org/3.3/library/stdtypes.html?highlight=maketrans#str.maketrans)/[`str.translate()`](https://docs.python.org/3.3/library/stdtypes.html?highlight=maketrans#str.translate).